### PR TITLE
Testing: Upgrade puppeteer to v2.0 to fix the broken interactive mode

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Reported by @diegohaz:

> Does anyone know why I can’t run `npm run test-e2e -- --puppeteer-interactive` to see the browser? I always get a `connect ECONNREFUSED 127.0.0.1:<PORT>` error. The Chrome browser opens and closes really fast.

This PR resolves the issues by upgrading the `puppeteer` to the new major version. As usual, new major version produces some warnings during `npm install` but we can't do anything about it since external packages need to update their `package.json` entries.

## How has this been tested?
`npm run test-e2e -- --puppeteer-interactive`

I confirmed it works with the latest version of `puppeteer`.

### CODE REVIEW

**Comment by youknowriad** [comment]
We should use the new emulateMedia features https://github.com/GoogleChrome/puppeteer/releases to remove our "disable animations plugiin" and also the environment variable we use to disable animations. Instead we should just put the browser in "prefers-reduced-motion" mode.

**Comment by gziolo** [comment]
> We should use the new emulateMedia features https://github.com/GoogleChrome/puppeteer/releases to remove our "disable animations plugiin" and also the environment variable we use to disable animations. Instead we should just put the browser in "prefers-reduced-motion" mode.

Can you file a new issue? I don't want to block this PR as you can't debug code at the moment ...

**Comment by youknowriad** [comment]
Definitely not a blocker, just something I thought of when reading the changelog. I might give it a try after this PR get merged,  it should be pretty easy.

**Comment by gziolo** [comment]
> Definitely not a blocker, just something I thought of when reading the changelog. I might give it a try after this PR get merged, it should be pretty easy.

You probably also know best how to tackle it. Ping me when you have PR ready, I can review pretty quick 👍

**Comment by gziolo** [comment]
It looks like I upgraded `axe` and they introduced or updates some rules which cause some tests to fail. I will have to silence them for now, similar to other rules which Gutenberg violates. The rules are:
- `aria-hidden-focus`
- `aria-input-field-name`

**Comment by gziolo** [comment]
> It looks like I upgraded `axe` and they introduced or updates some rules which cause some tests to fail. I will have to silence them for now, similar to other rules which Gutenberg violates. The rules are:
> 
> * `aria-hidden-focus`
> * `aria-input-field-name`

Done with 8de7f70, added to the list of rules to enable in #15452.

**Review by ntwb — APPROVED** [review]
👍🏼

**Comment by afercia** [comment]
Thanks @gziolo! Regarding `aria-hidden-focus` please see my comment on https://github.com/WordPress/gutenberg/issues/11631#issuecomment-548379876

Regarding `aria-input-field-name` can you please point me at the log or the reference where it happens?

**Comment by afercia** [comment]
Ah! Thanks. That seems to relate to the table block: the editable within the table cells need to be labelled. RIght not, there's nothing there to label them, just:
`<div role="textbox" .... `

Which, in ARIA terms, is equivalent to an input field with no label.

**Pull request merged**
